### PR TITLE
Added reverse lookup for ISO639 codes, from two-letter to three-letter codes, and vice versa

### PR DESCRIPTION
**Pull request by william1835**
Implemented this in response to https://github.com/goavki/phenny/issues/186. So, if the two-letter code is supplied, a three-letter code is given, and if a three-letter code is supplied, a two-letter code is given.

Also replaced old `%` format syntax with `.format()` syntax

### CODE REVIEW

**Comment by sushain97** [comment]
Would you mind git squashing your commits?

File: modules/iso639.py [hunk 1]
```diff
@@ -57,7 +57,14 @@ def iso639(phenny, input):
                     #phenny.say("%s %s %s" % (oneCode, oneLang.lower(), thisCode.lower()))
     if response == "":
         response = "Sorry, %s not found" % thisCode
-
+    elif len(thisCode) == 2:
+        response += ". ISO 639-3 is %s" % phenny.iso_conversion_data[thisCode]
```
Comment (sushain97) [inline modules/iso639.py]: Can you handle the case that `thisCode not in phenny.iso_conversion_data`?

File: modules/iso639.py [hunk 2]
```diff
@@ -57,7 +57,14 @@ def iso639(phenny, input):
...
+    elif len(thisCode) == 3:
+        thisCodeIso1 = ""
+        for iso1, iso3 in phenny.iso_conversion_data.items():
+            if thisCode == iso3:
+                thisCodeIso1 = iso1
```
Comment (sushain97) [inline modules/iso639.py]: Could you make this code a bit less verbose? :) Perhaps something like http://stackoverflow.com/a/11658633/1266600

File: modules/iso639.py [hunk 3]
```diff
@@ -57,7 +57,14 @@ def iso639(phenny, input):
...
+        response += ". ISO 639-1 %s %s" % (("is", thisCodeIso1) if thisCodeIso1 != "" else ("was","not found"))
```
Comment (sushain97) [inline modules/iso639.py]: So, let's actually go for a bit smaller output :) How about:

.iso639 hin
begiak: hin, hi = Hindi

.iso639 hi
begiak: hi, hin = Hindi

.iso639 sco
sco = Scots

We don't actually need to mention whether it exists or not, or what it is, just say them.

**Comment by william1835** [comment]
Sure, I'll get right on it

**Comment by william1835** [comment]
@sushain97 How do I squash commits?

File: modules/iso639.py [hunk 4]
```diff
@@ -55,6 +53,16 @@ def iso639(phenny, input):
                     else:
                         response = template % (oneCode, oneLang)
                     #phenny.say("%s %s %s" % (oneCode, oneLang.lower(), thisCode.lower()))
+        elif thisCode in phenny.iso_data:
+            altCode = ""
```
Comment (sushain97) [inline modules/iso639.py]: Use `altCode = None`. Then, you can do `if altCode` later.

File: modules/iso639.py [hunk 5]
```diff
@@ -55,6 +53,16 @@ def iso639(phenny, input):
...
+            if len(thisCode) == 2 and altCode in phenny.iso_conversion_data[thisCode]:
+                altCode = phenny.iso_conversion_data[thisCode]
+            elif len(thisCode) == 3:
+                for iso1, iso3 in phenny.iso_conversion_data.items():
+                    if thisCode == iso3:
+                        altCode = iso1
```
Comment (sushain97) [inline modules/iso639.py]: Can you break out of the loop here?

Comment (william1835) [inline modules/iso639.py]: Sure, I did have it in the first place, must have forgot to add it in

**Comment by sushain97** [comment]
Hm, per @wei2912 the Travis tests should be passing. Could you change the tests to conform with your new changes and/or point out what the error is if it's external to your work? Changing the tests would also count as making a transcript.

**Comment by william1835** [comment]
@sushain97 OK, made the changes now
Transcript of examples:
17:20 < Guest64413> .iso639 eng
17:20 < bot1234> eng, en = English
17:20 < Guest64413> .iso639 en
17:20 < bot1234> en, eng = English
17:20 < Guest64413> .iso639 sco
17:20 < bot1234> sco = Scots

**Comment by sushain97** [comment]
Follow up: rebasing may fix the tests failing.

**Comment by sushain97** [comment]
Transcript looks good.

**Comment by william1835** [comment]
@sushain97 Sure, will do

**Comment by wei2912** [comment]
Test failures aren't due to this PR. I have a minor change that needs to be made, but will approve this task so that you can start working on the test.

**Comment by william1835** [comment]
@wei2912 Yep, got it

**Comment by wei2912** [comment]
Thanks a lot for that commit! You can work on a test now, if you wish. Take a look at the tests we have so far (such as `test_apy.py`), and claim the task "write a set of tests for a begiak module".

**Comment by william1835** [comment]
@wei2912 No problem. Do you know why the Travis build is failing?

**Pull request merged**